datalad_next/itertools/reroute.py

```diff
@@ -0,0 +1,232 @@
+""" Functions that allow to route data around upstream iterator """
+
+from __future__ import annotations
+
+from typing import (
+    Any,
+    Callable,
+    Generator,
+    Iterable,
+)
+
+
+__all__ = ['dont_process', 'join_with_list', 'route_in', 'route_out']
+
+
+dont_process = object()
+
+
+def route_out(iterable: Iterable,
+              data_store: list,
+              splitter: Callable[[Any], tuple[Any, Any | None]],
```

Comment:
What is the meaning of the second item in the return tuple being allowed `None`? This seems undocumented.

Reply:
That is actually not necessary, since `Any` includes `None` and because the second element of the result-tuple is not inspected by `route_out`.

`None` is removed in eca443038f4a73edd52b4713d889636c55f8bfdf